src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -600,13 +604,15 @@ private Stream GetContentStream(Uri uri, out Encoding encoding)
             Stream stream = null;
             encoding = null;
             string failureReason = null;
+            IList<FileData> files = _currentV2Run.Files;
 
-            if (uri != null && _currentV2Run.Files != null)
+            if (uri != null && files != null && _fileKeyToIndexDictionary != null)
             {
-#if FILES_ARRAY_WORKS
-                FileData fileData;
-                if (_currentV2Run.Files.TryGetValue(uri.OriginalString, out fileData))
+                if (_fileKeyToIndexDictionary.TryGetValue(uri.OriginalString, out int index)
+                    && index < files.Count)
```

Comment:
> && index < files.Count [](start = 19, length = 23)

when would this index < files.Count not be true? that seems like an adverse conition

Reply:
You're right: it's guaranteed by the way the dictionary is constructed. Removed this condition.

---
In reply to: [247376623](https://github.com/Microsoft/sarif-sdk/pull/1211#discussion_r247376623) [](ancestors = 247376623)